dejafu-tests/lib/Examples/ParMonad/Direct.hs

```diff
@@ -445,14 +448,17 @@ rescheduleR cnt kont = do
   mtask  <- lift $ popWork mysched
   case mtask of
     Nothing -> do
-                  Session _ finRef:_ <- lift $ readIORef $ sessions mysched
-                  fin <- lift $ readIORef finRef
-                  if fin
-                   then kont (error "Direct.hs: The result value from rescheduleR should not be used.")
-                   else do
-                     lift $ steal mysched
-                     lift   yield
-                     rescheduleR (cnt+1) kont
+                  sessions <- lift $ readIORef $ sessions mysched
+                  case sessions of
+                    Session _ finRef:_ -> do
+                      fin <- lift $ readIORef finRef
+                      if fin
+                      then kont (error "Direct.hs: The result value from rescheduleR should not be used.")
+                      else do
+                        lift $ steal mysched
+                        lift   yield
+                        rescheduleR (cnt+1) kont
+                    sessions -> error "sessions"
```

Comment:
Similarly here

Reply:
Sure thing. 

Any plans for a Hackage release ?

Reply:
Yes, I'll put one out today.